src/main/java/net/openhft/chronicle/threads/MediumEventLoop.java

```diff
@@ -520,22 +555,20 @@ protected void addNewHandler(@NotNull final EventHandler handler) {
             default:
                 throw new IllegalArgumentException("Cannot add a " + handler.priority() + " task to a busy waiting thread");
         }
-        if (thread != null)
+        if (thread == Thread.currentThread())
             handler.loopStarted();
     }
```

Comment:
Only call `loopStarted` if we're the event loop thread. This is equivalent to the previous condition given the protectections I added, but it exposes the intent better like this.